python-checks/src/main/java/org/sonar/python/checks/cdk/ResourceAccessPolicyCheck.java

```diff
@@ -32,29 +44,64 @@
 @Rule(key = "S6304")
 public class ResourceAccessPolicyCheck extends AbstractIamPolicyStatementCheck {
 
+  private static final Logger LOG = Loggers.get(ResourceAccessPolicyCheck.class);
   private static final String MESSAGE = "Make sure granting access to all resources is safe here.";
   private static final String SECONDARY_MESSAGE = "Related effect";
+  String resourceNameSensitiveAwsActions = "ResourceAccessPolicyCheck.txt";
+  Set<String> sensitiveAwsActions = null;
+
+  void init() {
+    try {
+      sensitiveAwsActions = new HashSet<>(loadResourceAsSet(resourceNameSensitiveAwsActions, StandardCharsets.UTF_8));
+    } catch (IOException e) {
+      sensitiveAwsActions = new HashSet<>();
+      LOG.error("Couldn't load resource '" + resourceNameSensitiveAwsActions + "', rule [S6304] ResourceAccessPolicyCheck will be disabled.", e);
+    }
+  }
+
+  @Override
+  public void initialize(SubscriptionCheck.Context context) {
+    super.initialize(context);
+    init();
+  }
+
+  static List<String> loadResourceAsSet(String resourceName, Charset charset) throws IOException {
+    try (InputStream is = ResourceAccessPolicyCheck.class.getResourceAsStream(resourceName)) {
+      if (is == null) {
+        throw new IOException("Cannot find resource file '" + resourceName + "'");
```

Comment:
Minor: I feel like we can avoid throwing this exception. We could instead directly call a method:
```java
void handleMissingResource() {
      sensitiveAwsActions = new HashSet<>();
      LOG.error("Couldn't load resource '" + resourceNameSensitiveAwsActions + "', rule [S6304] ResourceAccessPolicyCheck will be disabled.", e);
}
```

Anyway, feel free to ignore this comment if you prefer otherwise

Reply:
Even if I remove this throw exception, there is still possibles `IOException` with the `instanciation` of the `InputStreamReader` / `BufferedReader`, and also with the `readLine()` call.

Reply:
sure, I'm fine with `loadResourceAsSet` still raising the exception for `BufferedReader` and `readLine()` calls. However it feels a bit odd to me creating a new exception when the `InputStream` is null, just to be then caught at the call site a few lines above

Reply:
Btw I just noticed `loadResourceAsSet` returns a `List`. Maybe we can just rename it `loadResource` to avoid confusion